22_pipeslab/pipeslab.txt

```diff
@@ -3,19 +3,106 @@ For each of the following, type a single command line that performs:
+maxim Videos
+maxim wildcards
+
+  
   3. Print the number of lines from "ls -l" which represent paths that were modified today (no matter which day we run the command)
   Answer: 
+  
+maxim@maxim-virtual-machine:~$ ls -l | grep $(date | cut -d' ' -f3)
+-rw-rw-r-- 1 maxim maxim  151 Dec 29 09:28 cow
+-rw-rw-r-- 1 maxim maxim  661 Dec 29 09:36 fiels
+-rw-rw-r-- 1 maxim maxim    0 Dec 29 16:25 ipconfig
+-rw-rw-r-- 1 maxim maxim    3 Dec 29 09:35 log
+-rw-rw-r-- 1 maxim maxim   50 Dec 29 09:40 text
+
 
   4. Modify (1) and (2) to count files only (no directories)
   Answer:
 
-  5. The command /sbin/ifconfig shows information regarding the machine's network interfaces. Use the command to print only the IP address of the machine. Remember not to show the special loopback address 127.0.0.1
+ 1. 
+ maxim@maxim-virtual-machine:~$ ls -la | grep "^-"
+-rw-------  1 maxim maxim 28728 Dec 29 21:32 .bash_history
+-rw-r--r--  1 maxim maxim   220 Dec 18 17:00 .bash_logout
+-rw-r--r--  1 maxim maxim  3637 Dec 18 17:00 .bashrc
+-rw-rw-r--  1 maxim maxim   151 Dec 29 09:28 cow
+-rw-r--r--  1 maxim maxim    25 Dec 18 17:11 .dmrc
+-rw-rw-r--  1 maxim maxim   661 Dec 29 09:36 fiels
+-rwxrwxr-x  1 maxim maxim   126 Dec 29 17:03 hw.7
+-rw-------  1 maxim maxim  1512 Dec 20 14:10 .ICEauthority
+-rw-rw-r--  1 maxim maxim     0 Dec 29 16:25 ipconfig
+-rw-------  1 maxim maxim    61 Dec 21 12:06 .lesshst
+-rw-rw-r--  1 maxim maxim     3 Dec 29 09:35 log
+-rw-r--r--  1 maxim maxim   675 Dec 18 17:00 .profile
+-rw-r--r--  1 maxim maxim     0 Dec 19 16:34 .sudo_as_admin_successful
+-rw-rw-r--  1 maxim maxim    50 Dec 29 09:40 text
+-rw-------  1 maxim maxim    66 Dec 20 14:09 .Xauthority
+-rw-------  1 maxim maxim  1156 Dec 26 17:22 .xsession-errors
+-rw-------  1 maxim maxim  1441 Dec 20 13:11 .xsession-errors.old
+
+All the files in the directory : hw.7 is executable
+  
+  
+Answer:
+
+maxim@maxim-virtual-machine:~$ ls -la | grep "^-" | grep '^...x'
+-rwxrwxr-x  1 maxim maxim   126 Dec 29 17:03 hw.7
+
+
+
+2. i made another file with xxx permission: file.test
 
+maxim@maxim-virtual-machine:~$ ls -la | grep "^-" | grep '^...x..x..x'
+-rwxrwxrwx  1 maxim maxim     0 Jan  2 20:24 file.test
+-rwxrwxr-x  1 maxim maxim   126 Dec 29 17:03 hw.7
+
+
+  
+  5. The command /sbin/ifconfig shows information regarding the machine's network interfaces. Use the command to print only the IP address of the machine. 
+  Remember not to show the special loopback address 127.0.0.1
+
+
+maxim@maxim-virtual-machine:~$ ifconfig | egrep '([0-9]{1,3}\.){3}[0-9]{1,3}' | grep Bcast | cut -d ':' -f 2 | cut -d' ' -f 1                                           10.61.40.39
+10.61.40.39
+
+  
 6. Joe wanted to sort a text file in place so he tried to run the command "sort myfile.txt > myfile.txt". What happened to the file? Why?
 
+Blank file after running the command
+second form running the command, filename doesn't delete anything from the file - it only creates an empty file if one did not exist, 
```

Comment:
bash will process the redirection first, then it executes the command...so by the time ```sort``` looks at the file, it's empty!